eclair-node-gui/src/main/scala/fr/acinq/eclair/gui/GUIUpdater.scala

```diff
@@ -79,16 +79,16 @@ class GUIUpdater(mainController: MainController) extends Actor with ActorLogging
       runInGuiThread(() => mainController.channelBox.getChildren.addAll(root))
       context.become(main(m + (channel -> channelPaneController)))
 
-    case ChannelRestored(channel, peer, remoteNodeId, isFunder, channelId, currentData) =>
+    case ChannelRestored(channel, peer, remoteNodeId, isFunder, currentData: Commitments) => // We are only interested in normal channel Commitments here
```

Comment:
Same, is it really necessary? Do HC need to emit a `ChannelRestored` event in the first place?

Reply:
Like with `ChannelRelayer`, HC also needs to inject itself into `Register` which listens to `ChannelRestored` event. This is again an approach where anything except stuff known at core level is ignored.